src/obj_proc.cpp

```diff
@@ -1909,6 +1913,83 @@ int pull_proc(struct char_data*ch, struct obj_data *obj, int cmd, char*arg, CHAR
 
    return eSUCCESS;
 }
+
+int szrildor_pass(struct char_data *ch, struct obj_data *obj, int cmd, char *arg, CHAR_DATA *invoker)
+{
+        struct obj_data *p;
+        // 30097
+        if (cmd) return eFAILURE;
+
+        if (obj->obj_flags.timer == 1800)
+        {       // Just created - check if this is the first pass in existence and if so, repop zone 161
+                bool first = TRUE;
+                for (p = object_list; p;p = p->next)
+                {
+                        if (obj_index[p->item_number].virt == 30097 && p != obj && p->obj_flags.timer != 1800)  // if any exist that are not at 1800 timer
+                        {
+                                first = FALSE;
+                                break;
+                        }
+                }
+                if (first && real_room(30000) != -1)
+                {
+                        reset_zone(world[real_room(30000)].zone);
+                }
+
+        }
+
+        obj->obj_flags.timer--;
+        struct obj_data *n;
+        if (obj->obj_flags.timer<=0)
+        {
+                // once one expires, ALL expire.
```

Comment:
This would suck for a second group/person that bought pass(es) after a group had already started the adventure.   Do we have anything in place to keep them from buying a pass that they can't use?    We could have pass giver only do it if A) there are none in the world (first one) or B) if there are < 5 in the world AND the requestor is in the same group as someone else that has one.     This isn't perfect, but it would take care of the issue.

Reply:
I like Brian's suggestion here.  I can write a bit of dialogue for the Undertaker where he explains you can invite additional people to help, but they need to be in a group with you.  I can also tell you with some degree of certainty that a group containing less than 4 players is going to have a real bad time in there.  Groups of 2 would just be straight-up annihilated unless they specifically knew where not to go.